stat/roc.go

```diff
@@ -126,3 +126,32 @@ func ROC(cutoffs, y []float64, classes []bool, weights []float64) (tpr, fpr, thr
 
 	return tpr, fpr, cutoffs
 }
+
+// TOC returns the Total Operating Characteristic for the classes provided,
+// which should be a boolean variable ordered by a rank variable. TOC also
+// returns the minimum and maximum bounds for the TOC.
+//
```

Comment:
// The input classes should be sorted according to the rank variable, if this is the case, the toc values can be interpreted as true positives where values above or equal to the given rank are assigned class true for each given rank from 1 to len(classes)+1.

Reply:
// SortWeightedLabeled can be used to sort classes together with the rank variable.

Reply:
Also, could include weights, in which case copy comment for SortWeightedLabeled from ROC.

Reply:
Yeah, I was wondering about that. How would be the best way to include weights in this?

Reply:
I can have a go at it if you like, and then send the changes to your branch?

Reply:
I just had a go and it gives unintuitive results for the max bound. I put the commit, but I'm not sure.

Reply:
It is working. I initially had the approach wrong.